### ISSUE TITLE
Enhancement: [rule-name] <a short description of my proposal>

### ISSUE THREAD

**Issue by szku01** [typescript-eslint/typescript-eslint#7845]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention/

### Description

I would like to mix `error` and `warn` for various selectors in the **naming-convention** rule.  
For example I'd like to allow (with a warning) PascalCase inside _objects_ (because a certain API endpoint expects those), but outright forbid them in _function names_.

The first SO question is actaully the same (not mine) when I search for the rule: https://stackoverflow.com/a/73112658

I'm not sure if it's possible to do in a sane fashion with how the current configuration works (warn/error as first param, options second), _maybe_ creating aliases for the various selectors (that can be used as overrides) can work:

- `@typescript-eslint/naming-convention-memberlike`
- `@typescript-eslint/naming-convention-variableLike`
- etc.

### Fail

```typescript
const user = { Name: 'John Doe' }; // warn
function GetUser() { return user; } // error
```

### Pass

```typescript
const user = { Name: 'John Doe' }; // unchanged
function getUser() { return user; } // fixed
```

**Comment by bradzacher** [comment]
Unfortunately ESLint doesn't support multiple reporting levels per rule - each rule has exactly one level that you configure.

The only way to achieve this is to export multiple rules.

There is a lot of maintenance burden in us considering something like this - which is just not worth it to support a niche usecase.

You can do this locally pretty easily - if you create a local plugin you can import our plugin and export the `naming-convention` rule as many times as you like with different aliases so you can configure the levels independently.
